packages/mongodb-dialects/java-driver/src/main/kotlin/com/mongodb/jbplugin/dialects/javadriver/glossary/PsiMdbTreeUtil.kt

```diff
@@ -270,24 +270,30 @@ fun PsiElement.findMongoDbCollectionReference(): PsiExpression? {
 /**
  * Resolves to the value of the expression if it can be known at compile time
  * or null if it can only be known at runtime.
+ *
+ * @return Pair<Boolean, Any?> A pair where the first component represents whether
+ * the value was resolved during compile time or not and the second component
+ * represents the resolved value itself
  */
-fun PsiElement.tryToResolveAsConstant(): Any? {
+fun PsiElement.tryToResolveAsConstant(): Pair<Boolean, Any?> {
     if (this is PsiReferenceExpression) {
         val varRef = this.resolve()!!
         return varRef.tryToResolveAsConstant()
-    } else if (this is PsiLocalVariable) {
-        return this.initializer?.tryToResolveAsConstant()
+    } else if (this is PsiLocalVariable && this.initializer != null) {
+        this.initializer!!.tryToResolveAsConstant()
```

Comment:
Watch out here, this should return, right? If we don't add a return here it will return false to null and ignore the expression in line 283.

Reply:
oh no 😬 yea I will make this change, thank you for catching 🙌